tests/task_critical_region_test.toit

```diff
@@ -29,3 +30,14 @@ test_cancel_can_timeout:
       with_timeout --ms=1:
         critical_do:
           sleep --ms=10000
+
+test_deadline_in_critical:
+  with_timeout --ms=100:
+    expect_not_null task.deadline
+    critical_do:
+      expect_not_null task.deadline
```

Comment:
Maybe also add a test that the timeout triggers,
```
  expect_throw "DEADLINE_EXCEEDED": 
    with_timeout --ms=10:
      critical_do:
        while true: yield

Reply:
It should not. If it does it will throw.

Reply:
I've added a different test case (with sleep) that triggers a timeout.